Main.py

```diff
@@ -9,7 +9,16 @@
 from ExtractMainContent import ExtractMainContent
 import sys
 import codecs
-
+import signal
+"""
+this function handles signal SIGINT generated when user presses ctrl-c
+if SIGINT is not handled manually abnormal termination occurs
+"""
+def signal_handler(signal, frame):
+    print('\nExiting gracefully')
+    sys.exit(0)
+#registering the signal SIGINT with signal_handler function    
+signal.signal(signal.SIGINT, signal_handler)

```

Comment:
can you register the signal in App function instead of doing it at the global level.
Don't forget to place the comment there?